src/v2i-hub/CARMAStreetsPlugin/src/J2735MapToJsonConverter.cpp

```diff
@@ -0,0 +1,267 @@
+
+
+#include "J2735MapToJsonConverter.h"
+
+namespace CARMAStreetsPlugin
+{
+
+    void J2735MapToJsonConverter::convertJ2735MAPToMapJSON(const std::shared_ptr<MapData> mapMsgPtr, Json::Value &mapJson) const
+    {
+        // Construct metadata
+        Json::Value metadata;
+        auto timestamp_utc = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+        metadata["timestamp"] = std::to_string(timestamp_utc);
+        mapJson["metadata"] = metadata;
+
+        // Construct Map Data
+        Json::Value mapDataJson;
+        mapDataJson["layer_id"] = mapMsgPtr->layerID;
+        mapDataJson["msg_issue_revision"] = std::to_string(mapMsgPtr->msgIssueRevision);
+        mapDataJson["layer_type"] = std::to_string(*mapMsgPtr->layerType);
+
+        // Construct intersections
+        const IntersectionGeometryList *intersections = mapMsgPtr->intersections;
+        Json::Value intersectionsJson;
+        // Assume there is only one intersection geometry for each intersection
+        for (size_t i = 0; i < intersections->list.count; i++)
+        {
+            Json::Value intersectionJson;
+            if (intersections->list.array != nullptr)
+            {
+                auto intersection = intersections->list.array[i];
+                intersectionJson["id"]["id"] = std::to_string(intersection->id.id);
+                intersectionJson["lane_width"] = std::to_string(*intersection->laneWidth);
+                intersectionJson["revision"] = std::to_string(intersection->revision);
+                intersectionJson["ref_point"]["lat"] = std::to_string(intersection->refPoint.lat);
+                intersectionJson["ref_point"]["long"] = std::to_string(intersection->refPoint.Long);
+                intersectionJson["ref_point"]["elevation"] = std::to_string(*intersection->refPoint.elevation);
+
+                // Convert Laneset
+                Json::Value laneSetJson;
+                convertLanesetToJSON(intersection, laneSetJson);
+                intersectionJson["lane_set"] = laneSetJson;
+            }
+            intersectionsJson["intersection_geometry"] = intersectionJson;
+        }
+        mapDataJson["intersections"] = intersectionsJson;
+        mapJson["map_data"] = mapDataJson;
+    }
+
+    void J2735MapToJsonConverter::convertLanesetToJSON(const IntersectionGeometry *intersection, Json::Value &laneSetJson) const
+    {
+        // Construct laneset
+        const auto laneSet = intersection->laneSet;
+        if (laneSet.list.array != nullptr)
```

Comment:
Does array not have an empty method?

Reply:
When there is no laneSet at all from this MAP object, this is the checking. There is no empty function here. Only count to return the number of element. But it will throw coredump if the laneSet does not exist in the intersection, which is not the case for most valid MAP message.